py3status/modules/conky.py

```diff
@@ -0,0 +1,448 @@
+]
+
+nvidia
+[
+    {'full_text': 'MTR Freq '}, {'full_text': '1390 ', 'color': '#adff2f'},
+    {'full_text': 'Perf '}, {'full_text': '1 ', 'color': '#adff2f'},
+    {'full_text': 'Mem Perc '}, {'full_text': '61', 'color': '#adff2f'},
+]
+
+bar
+[
+    {'full_text': '#.... ', 'color': '#ffffff'},
+    {'full_text': '##... ', 'color': '#00FF00'},
+    {'full_text': '###.. ', 'color': '#FFA500'},
+    {'full_text': '####. ', 'color': '#FFFF00'},
+    {'full_text': '#####', 'color': '#FF0000'},
+]
+"""
+
+from subprocess import Popen, PIPE, STDOUT
+from threading import Thread
+from tempfile import NamedTemporaryFile
+from json import dumps
+from os import remove as os_remove
+
+STRING_NOT_INSTALLED = "not installed"
+STRING_MISSING_FORMAT = "missing format"
+
+
+class Py3status:
+    """
+    """
+
+    # available configuration parameters
+    config = {}
+    format = None
+    thresholds = []
+
+    def post_config_hook(self):
+        if not self.py3.check_commands("conky"):
+            raise Exception(STRING_NOT_INSTALLED)
+        elif not self.format:
+            raise Exception(STRING_MISSING_FORMAT)
+
+        # placeholders
+        self.delimiter = getattr(self, "delimiter", ".")
+        placeholders = self.py3.get_placeholders_list(self.format)
+        _placeholders = [x.replace(self.delimiter, " ") for x in placeholders]
+        colors = self.py3.get_color_names_list(self.format)
+        _colors = []
+        for color in colors:
+            if not getattr(self, "color_{}".format(color), None):
+                _colors.append(color.replace(self.delimiter, " "))
+        self.placeholders = placeholders + colors
+        conky_placeholders = _placeholders + _colors
+
+        # init
+        self.cache_names = {}
+        self.thresholds_init = colors
+        self.config.update({"out_to_x": False, "out_to_console": True})
+        self.separator = "|SEPARATOR|"  # must be upper
+
+        # make an output. TODO: make a better lua->python conversion
+        config = dumps(self.config, separators=(",", "=")).replace('"', "")
+        text = self.separator.join(["${%s}" % x for x in conky_placeholders])
+        tmp = "conky.config = {}\nconky.text = [[{}]]".format(config, text)
+
+        # write tmp output to '/tmp/py3status-conky_*'
+        self.tmpfile = NamedTemporaryFile(
+            prefix="py3status_conky-", suffix=".conf", delete=False
+        )
+        self.tmpfile.write(tmp if self.py3.is_python_2() else str.encode(tmp))
+        self.tmpfile.close()
+
+        # make a command
+        self.conky_command = "conky -c {}".format(self.tmpfile.name).split()
+
+        # thread
+        self.line = ""
+        self.error = None
+        self.process = None
+        self.t = Thread(target=self._start_loop)
+        self.t.daemon = True
+        self.t.start()
+
+    def _cleanup(self):
+        self.process.kill()
+        if getattr(self, "cleanup", True):
```

Comment:
Another hidden config?

Reply:
Yes on undocumented hidden trivial config. It's there just in case we need to troubleshoot something. This shouldn't be hardcoded.